pkg/controller/hyperconverged/hyperconverged_controller_test.go

```diff
@@ -207,13 +212,17 @@ var _ = Describe("HyperconvergedController", func() {
 				Expect(hco.Status.RelatedObjects).To(ContainElement(*objectRef))
 			})
 
-			It("should update if outdated", func() {
+			It("should update if outdated when in upgrade mode", func() {
+				// TODO: add negative test on the value we want to preserve
 				expectedResource := newKubeVirtConfigForCR(hco, namespace)
 				expectedResource.ObjectMeta.SelfLink = fmt.Sprintf("/apis/v1/namespaces/%s/dummies/%s", expectedResource.Namespace, expectedResource.Name)
 				outdatedResource := newKubeVirtConfigForCR(hco, namespace)
 				outdatedResource.ObjectMeta.SelfLink = fmt.Sprintf("/apis/v1/namespaces/%s/dummies/%s", outdatedResource.Namespace, outdatedResource.Name)
 				outdatedResource.Data[virtconfig.SmbiosConfigKey] = "old-smbios-value-that-we-have-to-update"
 				outdatedResource.Data[virtconfig.MachineTypeKey] = "old-machinetype-value-that-we-have-to-update"
+				// force upgrade mode
+				hco.Status.UpdateVersion(hcoVersionName, OldVersion)
+				os.Setenv(util.HcoKvIoVersionName, NewVersion)
```

Comment:
I think we need to add a new test case w/o upgrade. Is that what you meant in the comment on line 216?

Reply:
done